src/js/AppContainer.jsx

```diff
@@ -0,0 +1,45 @@
+var React = require('react');
+var api = require('./api');
+var UserSelect = require('./UserSelect.jsx');
+var UserFiHeading = require('./UserFiHeading.jsx');
+
+var AppContainer = React.createClass({
+
+  getInitialState: function(){
+    return {
+      user: {
+        name: '',
+        institutions: []
+      }
+    }
+  },
+
+  selectCallback: function(e){
+    var user = e.target.value;
+    var self = this;
+    console.log('selecting user: ', user);
+    api.getInstitutions(user, function(institutions){
+      self.setUser({name: user, institutions: institutions});
+    });
+  },
+
+  setUser: function(user){
+    this.setState({user: user});
+  },
+
+  render: function() {
+    return (
+      <div>
+        <UserSelect callback={this.selectCallback}/>
+        <UserFiHeading institution={{}} year="2017" user={this.state.user.name}/>
```

Comment:
Hmm, I think I see what you mean.

But the only time that we care to render the institution name is on any page but the home page. So, maybe its ok this way? We render the `UserFiHeading` no matter what, but it gets sent an institution if `this.props.params.institution` is set or it gets sent the default empty object, `{}` like it does now.

Or am I missing something else?